src/core/server/http/http_server.ts

```diff
@@ -233,4 +266,21 @@ export class HttpServer {
     // https://github.com/hapijs/hapi/blob/master/API.md#-serverauthdefaultoptions
     this.server.auth.default('session');
   }
+  private getAuthData(request: KibanaRequest | Request) {
+    const incomingMessage =
+      request instanceof KibanaRequest ? request.unstable_getIncomingMessage() : request.raw.req;
+
+    const hasState = this.authState.has(incomingMessage);
+    const state = this.authState.get(incomingMessage);
+    const status: AuthStatus = hasState
+      ? AuthStatus.authenticated
+      : this.authRegistered
+      ? AuthStatus.unauthenticated
+      : AuthStatus.unknown;
+
+    return { status, state };
+  }
+  private isAuthenticated(request: KibanaRequest | Request) {
```

Comment:
**note**: HttpServer in the current state is mix of infrastructure layer & application layer logic, we can restructure it later, as soon as finish with Security integration.